[RB9823] iP

bnahor opened the pull request:
## DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

### Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
A-TextUiTesting

Commit:
Level-5

Commit:
Level-7

J-hta-n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,106 @@
+        actionList.add(loadedTasks);
+    }
+
+    public void initiateChatbot() {
```
Would it be possible to shorten this method (and perhaps reduce the depth of nesting)?

zhyuhan replied:
You can consider adding some line breaks to make your code more readable.

J-hta-n commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,41 @@
+        public static Command translate(String input) {
+            String lowerInput = input.toLowerCase();
+            String command = lowerInput.split(" ")[0];
+            switch (command) {
```
Minor issue with the case indent

J-hta-n commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,42 @@
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    public String getDescription() {
```
Is this method necessary?

J-hta-n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,106 @@
+                                "Format: event task /from date|time /to date|time");
+                    }
+                    String[] commandParts = parser.splitByKeyword(inputParts[1], "/from");
+                    String[] eventParts = parser.splitByKeyword(commandParts[1], "/to");
```
I like how the naming of the parsed commands and arguments are clear and adhere well to the plural form for all arrays 👍

J-hta-n commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,106 @@
+import java.util.ArrayList;
+import java.util.Scanner;
 public class Duke {
```
Perhaps can include some comments to explain what each component does and why (javadocs for classes and key methods) as well?

J-hta-n commented on src/main/java/Save.java:
```diff
@@ -0,0 +1,68 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Save {
```
I like how the code is concise and readable with proper naming of methods and variables 👍

J-hta-n reviewed (commented):
- Overall, code base is concise and naming guidelines are followed well which allowed for readability 👍 
- Just a few pointers regarding javadocs and code quality :)

zhyuhan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,42 @@
+        this.isDone = true;
+    }
+
+    public void unMark() {
```
Perhaps you could change this method name to `unmark`?

zhyuhan commented on src/main/java/Save.java:
```diff
@@ -0,0 +1,68 @@
+            File directory = new File(DIRECTORY_PATH);
+            if (!directory.exists()) {
+                directory.mkdirs();
+            } try (BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH))) {
```
Seem like this inner `try` is missing a `catch` block.

zhyuhan replied:
Perhaps you could use a single `try-catch` and handle the any exceptions separately.

zhyuhan commented on src/main/java/Save.java:
```diff
@@ -0,0 +1,68 @@
+        switch (parts[0]) {
+            case "T":
+                Todo todo = new Todo(parts[2]);
+                if (parts[1].equals("1")) todo.markAsDone();
```
This conditional should be on separate lines (see [here](https://se-education.org/guides/conventions/java/intermediate.html#conditionals)).

zhyuhan commented on src/main/java/Save.java:
```diff
@@ -0,0 +1,68 @@
+    private static Task parseLineToTask(String line) {
+        String[] parts = line.split(" \\| ");
+        switch (parts[0]) {
+            case "T":
```
Each `case` should have no indent (see [here](https://se-education.org/guides/conventions/java/intermediate.html#layout)).

zhyuhan commented on src/main/java/UI.java:
```diff
@@ -0,0 +1,25 @@
+import java.util.Scanner;
+public class UI {
+    private final Scanner scanner = new Scanner(System.in);
+    public void horizontalLines() {
```
The method name should be a verb.

zhyuhan commented on src/main/java/Actions.java:
```diff
@@ -0,0 +1,53 @@
+        for (int i = 0; i < actions.size(); i++) {
+            String formattedIndex = String.format("%2d", i + 1);
+            formattedTasks.add(formattedIndex + ". " + actions.get(i));
+        } return String.join("\n", formattedTasks);
```
Adding a newline here would make the code much more readable!

zhyuhan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,106 @@
+                case LIST:
+                    ui.lineSandwich(actionList.stringList());
+                    break;
+                case MARK:
```
You should consider adding a comment to explicitly say this case has a fallthrough.

bnahor committed:
Level-8

bnahor committed:
A-MoreOOP

bnahor committed:
A-Packages

bnahor committed:
Extra

bnahor committed:
Merge branch 'master' into add-gradle-support

bnahor committed:
gradle support fixed

bnahor committed:
JUnit Tests Added

bnahor committed:
A-CodingStandard

bnahor committed:
Level-9 fixed

bnahor committed:
JavaDocs and Exception Handling for all Methods

bnahor committed:
All classes: Add JavaDocs and Exception Handling

JavaDocs provide summary of method functionality and input/return
parameters for easier identification.

Exception handling allows for more defined program behaviour, with
proper responses for unintended inputs by the user. It also enables the
user to better use the application.

bnahor committed:
Merge branch 'branch-A-CodingStandard'

bnahor committed:
Rectify Exception Messages and JavaDoc Formatting

bnahor committed:
Add GUI

bnahor committed:
Modify GUI parameters

bnahor committed:
Modify CLI Functionality

bnahor committed:
Add assert statements.

Assert statements help prevent errors by ensuring input / output
conditions are met.

bnahor committed:
Merge pull request #2 from RB9823/branch-A-Assertions

Add assert statements.

bnahor committed:
Improve Code Quality.

Aids in understanding the code fucntionality, which enables easier
additions / modifications.

bnahor committed:
Improve Code Quality

Handling of GUI improved with consistent output for list and schedule
commands.

Fix text wrapping for longer messages

bnahor committed:
Merge pull request #3 from RB9823/branch-A-CodeQuality

Update Code Quality

bnahor committed:
More Changes

bnahor committed:
Merge pull request #4 from RB9823/branch-A-CodeQuality

More Changes

bnahor committed:
hell

bnahor committed:
Extend Whelmed Functionality with Archive

Archive enables storage of a given list for later access, while starting
with a fresh list.

bnahor committed:
Add User Guide

bnahor committed:
Fix gradle.build
